scheduler/core/service/service.py

```diff
@@ -68,3 +76,90 @@ def build_scheduler(start: Time = Time("2018-10-01 08:00:00", format='iso', scal
     semesters = frozenset([Semester.find_semester_from_date(start.to_value('datetime')),
                            Semester.find_semester_from_date(end.to_value('datetime'))])
     return Service(start, end, semesters, sites)
+
+
+
+
+
+def _get_total_bands_by_program(programs):
+    band_1 = 0
+    band_2 = 0
+    band_3 = 0
+    band_4 = 0
+    for p in programs:
+        if program.band is Band.Band1:
+            band_1+=1
+        elif program.band is Band.Band2:
+            band_2+=1
+        elif program.band is Band.Band3:
+            band_3+=1
+        elif program.band is Band.Band4:
+            band_4+=1
+
+    return band_1, band_2, band_3, band_4
```

Comment:
This seems a little rough, especially returning `Tuple[int, int, int, int]`.

Could we instead do something like:
```python
# List or Set?
def _get_total_bands_by_program(programs: List[Program]) -> Dict[Band, int]:
    bands: Dict[Band, int] = {}
    for p in programs:
        bands[p.band] = bands.setdefault(p.bands, 0) + 1
    return bands
```

Reply:
Maybe `Iterable` instead of `List` or `Set`... that seems to make more sense.